src/main/java/ru/herobrine1st/fusion/module/vk/util/Extensions.kt

```diff
@@ -53,6 +54,21 @@ fun Post.toEmbeds(wallName: String, wallAvatarUrl: String?, repost: Boolean = fa
             }
             if (size > 4) footerBuilder.append("Post contains more than 4 images\n")
         }
+        with(attachments.filterIsInstance<Document>()) {
+            if (this.isEmpty()) return@with
+            with(this.filter { it.type.toDocumentType() == Document.Type.Gif }) {
+                embedBuilder.setImage(this[0].url)
+                this.subList(1, min(4, size)).forEach {
+                    embeds.add(
+                        ModifiedEmbedBuilder()
+                            .setTitle(it.title, it.url)
```

Comment:
ModifiedEmberBuilder is created as a workaround to allow title be null while setting url. Embed with both title and url is shown as distinct embed. Also url is set to url of a post, because it is a merge key.
```diff
-.setTitle(it.title, it.url)
+.setTitle(null, url)
```